src/http_microhttpd.c

```diff
@@ -629,25 +635,29 @@ static int encode_and_redirect_to_splashpage(struct MHD_Connection *connection,
 static int redirect_to_splashpage(struct MHD_Connection *connection, t_client *client, const char *host, const char *url)
 {
 	char *originurl = NULL;
-	char query_str[QUERYMAXLEN] = {0};
-	char *query = query_str;
-	int ret = 0;
-	const char *separator = "&";
+	char query[QUERYMAXLEN] = { 0 };
 	char *querystr = NULL;
 	s_config *config = config_get_config();
 
-	get_query(connection, &query, separator);
-	if (!query) {
+	struct get_query_data data = {
+		.error = false,
+		.size = 0,
+		.capacity = QUERYMAXLEN,
+		.query = query,
+	};
+
+	// collect query
+	if (MHD_get_connection_values(connection, MHD_GET_ARGUMENT_KIND, get_query_callback, &data) < 0 || data.error) {
 		debug(LOG_DEBUG, "Unable to get query string - error 503");
-		/* no mem */
+		/* not enough memory */
 		return send_error(connection, 503);
```

Comment:
We should change this 503.
IMHO: the user is accessing a website with a long long query string (not sure what the spec says about the maximum length), but in the of redirection we should better go for dropping the whole query string.

Reply:
Agreed, we might trigger some unindentent action :D

Reply:
Hm, but that does not sound trivial. Maybe I should just drop that commit.

Reply:
This is a different position than your first commit. You can leave your first commit in and I'll do a different patch based on top of it.